awx/main/management/commands/enable_instance.py

```diff
@@ -0,0 +1,78 @@
+from urllib.parse import urljoin
+
+from argparse import ArgumentTypeError
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+
+from awx.main.models import Instance, UnifiedJob
+
+
+class AWXInstance:
+    def __init__(self, **filter):
+        self.filter = filter
+        self.get_instance()
+
+    def get_instance(self):
+        filter = self.filter if self.filter is not None else dict(hostname=settings.CLUSTER_HOST_ID)
+        qs = Instance.objects.filter(**filter)
+        if not qs.exists():
+            raise ValueError(f"No AWX instance found with {filter} parameters")
+        self.instance = qs.first()
+
+    def disable(self):
+        if self.instance.enabled:
+            self.instance.enabled = False
+            self.instance.save()
+            return True
+
+    def enable(self):
+        if not self.instance.enabled:
+            self.instance.enabled = True
+            self.instance.save()
```

Comment:
I don't want to give an opinion on the overall intent of this PR.

But if this command is added, it is fairly critical that it reduces the database contention problems by using `self.instance.save(update_fields=['enabled'])`. This is the only field it is expected to modify, and this avoids updating unrelated columns with stale data. We have a lot of problems with separate processes modifying instances around the same time.

Reply:
The same goes for `disable_instance` as well, since it uses `self.instance.save()` instead of specifying a specific field.